app/models/server/raw/Rooms.js

```diff
@@ -325,7 +326,7 @@ export class RoomsRaw extends BaseRaw {
 		const params = [...firstParams, sort];
 		if (onlyCount) {
 			params.push({ $count: 'total' });
-			return this.col.aggregate(params, { allowDiskUse: true });
+			return this.col.aggregate(params, { allowDiskUse: true, readPreference: ReadPreference.SECONDARY_PREFERRED });
```

Comment:
any reason to not use `readSecondaryPreferred`?

https://github.com/RocketChat/Rocket.Chat/blob/78b74c4cf7b5eab93b4a183052aaf7857aa9ed62/server/overrides/database/readSecondaryPreferred.ts#L3

Reply:
Not really, it was easier to pass the option to the query directly, but let me change to use the function

Reply:
Changed!